dwave/optimization/include/dwave-optimization/graph.hpp

```diff
@@ -158,6 +158,20 @@ class Graph {
     std::span<DecisionNode* const> decisions() noexcept { return decisions_; }
     std::span<const DecisionNode* const> decisions() const noexcept { return decisions_; }
 
+    // Remove unused nodes from the graph.
+    //
+    // This method will reset the topological sort if there is one.
+    //
+    // A node is considered unused if all of the following are true:
+    // * It is not a decision.
```

Comment:
This is currently a constraint because of the difficulty of removing some types of decisions, correct?

Reply:
It's not actually difficult to implement but there are a few reasons:
1) I can see how symbols might be accidentally added to the model, but for decisions I think there is a high chance they are there deliberately anyway. I.e. i doubt we'd be removing decisions often.
2) This way if one was to use this method as part of a presolve step, the serialized states of the model after presolve would still be relevant to the original model.

Reply:
Good point, I didn't think about the presolve angle.